### PR TITLE
feat: support v1/embeddings

### PR DESCRIPTION
**Pull request by Geniucker**

### CODE REVIEW

**Comment by Geniucker** [comment]
can you provide your request body

File: main.go [hunk 1]
```diff
@@ -222,6 +228,88 @@ func chatCompletions(c *gin.Context) {
...
+		return
+	}
+	// check if the input is a list, if not, wrap it in a list
+	if _, ok := jsonBody.Input.([]interface{}); !ok {
+		jsonBody.Input = []interface{}{jsonBody.Input}
+	}
+
+	jsonData, err := json.Marshal(jsonBody)
+	if err != nil {
+		return
```
Comment (funnyzak) [inline main.go:282]: 转换错误的时候，是不是可以 c.json 个提示并logging错误信息，方便调试

**Review by funnyzak — APPROVED** [review]

**Comment by Geniucker** [comment]
#160 I have no idea about that. If anyone can offer some help

**Review by aaamoon — APPROVED** [review]

**Comment by Geniucker** [comment]
P.S The embeddings API of gh copilot is not the same as the one of OpenAI. For copilot, only strings or list of strings are accepted while OpenAI accepts both int and string.

**Comment by Geniucker** [comment]
- `POST /v1/embeddings`
    - for embeddings api  
    Pay attention that this api is **not totally compatible** with OpenAI API.  
    For `input` field, OpenAI API accepts following types:
        - `string`: The string that will be turned into an embedding.
        - `array`: The array of strings that will be turned into an embedding.
        - `array`: The array of integers that will be turned into an embedding.
        - `array`: The array of arrays containing integers that will be turned into an embedding.

        Unfortunately, this service only accepts the first 2 types as well as the array of arrays containing strings.

**Comment by Geniucker** [comment]
I think I've done all I can do.

**Comment by thangckt** [comment]
does this change solve this issue #214

**Comment by Geniucker** [comment]
I don't think they are related

**Comment by Geniucker** [comment]
@funnyzak @aaamoon 这个PR是关了还是怎么说，毕竟和OpenAI的接口并不一致

**Review by funnyzak — APPROVED** [review]

**Comment by funnyzak** [comment]
我觉得，在文档说明下，合并吧。

**Review by funnyzak — APPROVED** [review]

**Comment by funnyzak** [comment]
本地 和 docker 简单测了下，没发现有什么问题。

**Comment by funnyzak** [comment]
@Geniucker @aaamoon 这个还有进一步修改吗？

**Comment by Geniucker** [comment]
没有了

**Comment by aaamoon** [comment]
可以先合了

**Comment by funnyzak** [comment]
合吧

**Comment by Geniucker** [comment]
我解决下冲突

**Comment by Chrisma-98** [comment]
> @funnyzak @aaamoon 这个PR是关了还是怎么说，毕竟和OpenAI的接口并不一致

LOL，It seems I found the problem. I'm using langchain and openai for text embedding. When using the API of this project as the backend, I found a BUG. Reviewing the request, I found: langchain now request the embedding address to `/v1/engines/models/embeddings` instead of '/v1/embeddings'! 0.O

**Comment by Geniucker** [comment]
Seems that langchain is using outdated APIs. Same for /v1/chat/completions

**Pull request merged**